### PR TITLE
add java solution cses-2079

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cses-2079.mdx [hunk 1]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+	for (int chi : g[cur]) {
+		if (chi != par) {
+			if (s[chi] * 2 > n) return get_centroid(chi, cur);
+		}
 	}
 	return cur;
 }
 
 int main() {
 	// input
```
Comment (envyaims) [inline solutions/cses-2079.mdx:46]:
```suggestion

```
we don’t need this lmao

**Review by envyaims — APPROVED** [review]
ok

File: solutions/cses-2079.mdx [hunk 2]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
 #include <bits/stdc++.h>
 using namespace std;
 
-const int maxn = 2e5+5;
-int n; // number of nodes
-vector<int> g[maxn]; // graph
-int s[maxn]; // size of subtree
+const int maxn = 2e5 + 5;
+int n;				// number of nodes
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:22]: what

File: solutions/cses-2079.mdx [hunk 3]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+vector<int> g[maxn];  // graph
+int s[maxn];		  // size of subtree
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:24]: yeah what

File: solutions/cses-2079.mdx [hunk 4]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
 </CPPSection>
 
+<JavaSection>
+
+```java 
+import java.io.*;
+import java.util.*;
+
+public class Centroid {
+	private static int n; // number of nodes
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:69]:
```suggestion
	private static int n;  // number of nodes
```
two spaces

File: solutions/cses-2079.mdx [hunk 5]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+	private static List<List<Integer>> graph;
+	private static int[] sub_array_size;
+	private static Kattio io;
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:72]: doesn't need to be static (`io`)

File: solutions/cses-2079.mdx [hunk 6]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+
+		for (int i = 0; i < n - 1; i++) {
+			int a = io.nextInt(), b = io.nextInt();
+			graph.get(a).add(b);
+			graph.get(b).add(a);
+		}
+
+		dfs_size(1, -1);
+
+		System.out.println(get_centroid(1, -1));
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:113]: you're using io, might as well
```suggestion
		io.println(get_centroid(1, -1));
```

File: solutions/cses-2079.mdx [hunk 7]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+		for (int next : graph.get(curr)) {
+			if (next == prev) continue;
+			dfs_size(next, curr);
+			sub_array_size[curr] += sub_array_size[next];
+		}
+	}
+
+	private static int get_centroid(int curr, int prev) {
+		// if possible, move towards the adjacent node with the subtree size greater
+		// than n / 2. Otherwise, we have found our centroid.
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:86]:
```java
/*
 * style multiline comments
 * like this
 */
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cses-2079.mdx (hunk 3 above)
Comment (envyaims) [reply solutions/cses-2079.mdx:24]: lmao i didn’t see that for some reason

Comment (876pol) [reply solutions/cses-2079.mdx:24]: L formatter

File: solutions/cses-2079.mdx (hunk 5 above)
Comment (876pol) [reply solutions/cses-2079.mdx:72]: I don't think it compiles without static because the main method is static

Comment (876pol) [reply solutions/cses-2079.mdx:72]: ill put the declaration in the main function

File: solutions/cses-2079.mdx [hunk 8]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+public class Centroid {
+	private static int n; // number of nodes
+	private static List<List<Integer>> graph;
+	private static int[] sub_array_size;
+
+	private static void dfs_size(int curr, int prev) {
+		// size of a subtree is the sum of the sizes of all child subtrees + 1.
+		sub_array_size[curr] = 1;
+		for (int next : graph.get(curr)) {
+			if (next == prev) continue;
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:76]: braces

File: solutions/cses-2079.mdx [hunk 9]
```diff
@@ -18,40 +18,107 @@ For more information about centroids and how to find/use them, see
...
+		}
+	}
+
+	private static int get_centroid(int curr, int prev) {
+		/*
+		 * if possible, move towards the adjacent node with the subtree size
+		 * greater than n / 2. Otherwise, we have found our centroid.
+		 */
+		for (int next : graph.get(curr)) {
+			if (next == prev) continue;
```
Comment (SansPapyrus683) [inline solutions/cses-2079.mdx:88]: same here

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**